lib/git-util.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+const execa = require('execa');
+
+exports.getLatestTag = async () => {
+	let latest;
+	try {
+		// In case a previous tag exists, we use it to compare the current repo status to.
+		latest = await execa.stdout('git', ['describe', '--abbrev=0']);
+	} catch (_) {
+		// Otherwise, we fallback to using the first commit for comparison.
+		latest = await execa.stdout('git', ['rev-list', '--max-parents=0', 'HEAD']);
+	}
+
+	return latest;
+};
```

Comment:
Sorry for being unclear. I meant the two Git commands here as separate methods. The `getLatestTag` naming is not really correct as it also returns a commit SHA when there are no tags.

How about 3 methods?

- `latestTag()`
- `firstCommit()`
- `latestTagOrFirstCommit()` (uses the first two)

Reply:
Note: I didn't expose the `latestTag` and `firstCommit` methods as they're not currently used anywhere accept for inside `lib/git-util.js`.